specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,203 @@
+    {
+
+        if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+        {
+            // capturing the mouse will allow us to begin forwarding messages 
+            // to webview
+            isCapturingMouse = true;
+            SetCapture(mainWindowHwnd);
+        }
+        else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+        {
+            // no longer need to capture the mouse for draggbale regions support 
+            // after this
+            isCapturingMouse = false;
+            ReleaseCapture();
+        }
+        
+        // forward mouse messages to WV
+        CompositionController.SendMouseInput(
+            (CoreWebView2MouseEventKind)message,
+            (CoreWebView2MouseEventVirtualKeys)GET_KEYSTATE_WPARAM(wParam),
+            mouseData,
+            point);
+
+        m.Result = 0
+        return;
+    }
+        
+}
+// The following code forwards mouse messages to the WebView2 - Start
+    ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+/// This enum contains values representing possible regions that Webview
+/// can support
+typedef enum COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT {
+    /// Caption region
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CAPTION,
+    /// Client region
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CLIENT,
+    /// Out of bounds of the app window
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_NONE
+} COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT;
+
+/// This interface includes the new API for enabling Webview support for hit-testing regions
+[uuid(42BF7BA5-917A-4C27-ADA1-EA6969854C16), object, pointer_default(unique)]
+interface ICoreWebView2StagingCompositionController4 : ICoreWebView2CompositionController3 {
+    /// Takes in a point and returns a Hit-test result value as an out parameter
+    /// point is in Screen Coordinates
+    [propget] HRESULT NonClientHitTestAtPoint(
+                        [in] POINT point, 
+                        [out, retval] COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT* val);
+}
+```
+## .Net/ WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core {
+    /// This enum contains values representing possible regions that Webview
+    /// can support
+    enum CoreWebView2NonClientHitTestResult
+    {
+        /// Caption region
+        Caption = 0,
+        /// Client region
+        Client = 1,
+        /// Out of bounds of the app window
+        None = 2,
+    }
+    /// This runtime class includes more than just the new API for enabling Webview support for hit-testing regions
+    runtimeclass CoreWebView2CompositionController {
+        
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2CompositionController4")]
+        {
+            /// Takes in a point and returns a Hit-test result value
+            /// point is in Screen Coordinates
+            CoreWebView2NonClientHitTestResult NonClientHitTestPoint(Windows.Foundation.Point point);
```

Comment:
This section will also need to be updated after you update the COM IDL above (this method name doesn't match anymore). (You don't have to include the reference documentation in this section if you don't want to - just the COM IDL section requires it)

Reply:
got it